Flow.Launcher.Infrastructure/StringMatcher.cs

```diff
@@ -50,7 +50,6 @@ public MatchResult FuzzyMatch(string query, string stringToCompare, MatchOption
 
             if (_alphabet != null)
             {
-                query = _alphabet.Translate(query);
```

Comment:
what's the reason for removing this? This should cater for the translation for Chinese to Pinyin.

Reply:
I moved that from using alphabet to program because I think query may not need to be in the cache. Also, Bao has assumed that query should be pinyin or English, so he removed it without adding it somewhere else. 
I agree with that but I think occasionally input chinese but results nothing may seems weird, so I moved it to program. 
It is also for removing extra translation since query may be translated in every fuzzy search, but if we moved it there, there will be only one translation. 

This seems a little bit ugly though😂.

Reply:
I would like to keep the ability to search using Chinese as well as Pinyin. In terms of it being called every time, only when user has Use Pinyin switched on. In terms of it being in memory cache, you can add a bool parameter in Translate that determines whether it should be stored.

What do you think?

Reply:
Well I mean that we will always do a fuzzy search between query and every single program in the program plugin, so that the translation will be called n times (number of program) once query change. However, if we put that in here, or somewhere the query is created, it will be called only one time per query. If we don't put it in memory cache, it may slow down the query if Chinese character is in the query.
I think the best way is to put it to the plugin manager or even earlier when the query is created and translate that if pinyin is enabled. I will take a look on that. That will enable every plugin to search with Chinese character.
Actually, I think that there won't be so much chance for us to search things with Chinese, because that is quite messy to type that in.

Reply:
I have taken a look on where can we put the translation for query. However, if we put it to when query is generated, whether that may break some characteristics of plugin that utilize Chinese search, and that's not what we want.

Therefore, I think maybe we should just keep it here, even though suffer a bit performance loss (though should not be significant). However, that seems not elegant hhhh. Let me think whether there may be a better way to solve that issue, because I think simply removing it or translating query separately is also not elegant.